### PR TITLE
Strided Array Support in Sort/Search Modules

### PR DESCRIPTION
**Pull request by ben-albrecht**
* Strided array support for Sort and Search module routines.
    * `Sort.quickSort()` and `Search.binarySearch()` overloaded with `where Dom.stridable` clause to preserve performance for non-strided arrays
* As a prerequisite for testing, added strided array support for `Random.shuffle()`
* Updated tests for `Search`, `Sort` and `Random.shuffle()` to confirm strided array support.

* *tested on linux64* with no failures

### CODE REVIEW

File: modules/packages/Search.chpl [hunk 1]
```diff
@@ -93,24 +97,56 @@ proc search(Data:[?Dom], val, comparator:?rec=defaultComparator, lo=Dom.low, hi=
...
   }
 
   return (false, Dom.high+1);
 }
 
 
+pragma "no doc"
+/* Strided linearSearch */
+proc linearSearch(Data:[?Dom], val, comparator:?rec=defaultComparator, lo=Dom.low, hi=Dom.high) 
+  where Dom.stridable {
```
Comment (mppf) [inline modules/packages/Search.chpl]: I'd expect you'd make the other linearSearch have a 'where !Dom.stridable'

File: modules/packages/Search.chpl [hunk 2]
```diff
@@ -152,6 +191,34 @@ proc binarySearch(Data:[?Dom], val, comparator:?rec=defaultComparator, in lo=Dom
 }
 
 
+pragma "no doc"
+/* Strided binarySearch */
+proc binarySearch(Data:[?Dom], val, comparator:?rec=defaultComparator, in lo=Dom.low, in hi=Dom.high) 
+  where Dom.stridable {
```
Comment (mppf) [inline modules/packages/Search.chpl]: same here

File: modules/packages/Sort.chpl [hunk 3]
```diff
@@ -573,6 +624,64 @@ proc quickSort(Data: [?Dom] ?eltType, minlen=16, comparator:?rec=defaultComparat
 }
 
 
+pragma "no doc"
+/*
+  Stridable quickSort
+ */
+proc quickSort(Data: [?Dom] ?eltType, minlen=16, comparator:?rec=defaultComparator) where Dom.stridable {
```
Comment (mppf) [inline modules/packages/Sort.chpl]: and here

**Comment by benharsh** [comment]
As we discussed offline, it's unfortunate that we have to duplicate some of these functions. For now though, it's worth getting this in so that we can preserve performance. We can clean things up later.

Looks good to me 👍

**Pull request merged**